File: 백준/Bronze/9506. 약수들의 합/약수들의 합.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

	public static void main(String[] args) throws IOException {

       BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
       
       while(true) {
    	   
    	   int num = Integer.parseInt(br.readLine());
    	   
    	   if(num == -1) {
    		   return;
    	   }
    	   
    	   int[] arr = new int[num];
    	   
    	   int sum = 0;
    	   int index = 0;
    	   
    	   for(int i = 1; i < num; i++) {
    		   if(num % i == 0) {
    			   arr[index++] = i;
    			   sum += i;
    		   }
    	   }
    	   
    	   if(sum != num) {
    		   System.out.println(num + " is NOT perfect.");
    	   } else if(sum == num) {
    		   System.out.print(num + " = ");
    		   for(int i = 0; i < index; i++) {
    			   System.out.print(arr[i]);
    			   if(i + 1 != index) {
    				   System.out.print(" + ");
    			   }
    			   
    		   }
    		   System.out.println();
    	   }
    	   
       }
       
	
	}


}

File: 백준/Bronze/9506. 약수들의 합/약수들의 합.cc
#include <iostream>
#include <vector>
using namespace std;

int main()
{	
	int n;
	while (true) {
		cin >> n;

		if (n == -1) break;

		vector<int> vec;
		int sum = 0;

		for (int i = 1; i < n; i++) {
			if (n % i == 0) vec.push_back(i);
		}

		for (int i = 0; i < vec.size(); i++) {
			sum += vec[i];
		}

		if (sum != n) {
			cout << n << " is NOT perfect." << "\n";
		}
		else if (sum == n) {
			cout << n << " = ";
			for (int i = 0; i < vec.size(); i++) {
				cout << vec[i];
				if ((i + 1) != vec.size())
					cout << " + ";
			}
			cout << "\n";
		}
		
	}
	return 0;
}